pkg/cloud/services/compute/instance.go

```diff
@@ -113,6 +104,25 @@ func (s *Service) CreateInstance(openStackCluster *infrav1.OpenStackCluster, mac
 	if err != nil {
 		return nil, err
 	}
+	// verify trunk support if trunk is set on at least one Port.
+	for _, net := range nets {
+		port := net.PortOpts
+		if port != nil {
+			// No Trunk field specified for the port, inherit openStackMachine.Spec.Trunk.
+			if port.Trunk == nil {
+				*port.Trunk = openStackMachine.Spec.Trunk
+			}
+			if openStackMachine.Spec.Trunk || *port.Trunk {
+				trunkSupported, err := s.isTrunkExtSupported()
+				if err != nil {
+					return nil, err
+				}
```

Comment:
Please can you move this outside the loop? This is adding an extra API call per port.

Reply:
When it was outside, it was checking only instance level before  `instanceSpec.Trunk = openStackMachine.Spec.Trunk`. Now, in the for loop, it is checking both instance level and port level.

I agree, in the current format, it is making API calls per port. I am trying to make it efficient by stopping at the first occurrence trunk-set port. I have not succeeded so far as combing `return` and `break` is not possible.

Note that making API calls per port is not avoidable. The other location where the check could be added is [here](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/180f320d048e5ed3797242a985dc511e4b24aabe/pkg/cloud/services/compute/instance.go#L135-L137). Both do iterate over the ports.

Reply:
The only thing you need to move outside the loop is the code above which sets trunkSupported. The value of trunkSupported will never change during the loop. You can leave the code below, which generates the error, inside the loop.

Reply:
Thanks. Moving it out.